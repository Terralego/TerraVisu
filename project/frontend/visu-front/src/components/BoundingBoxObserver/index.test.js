import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import VisibleBoundingBox from './BoundingBoxObserver';

jest.mock('resize-observer-polyfill', () => jest.fn(callback => (
  {
    observe: jest.fn(() => {
      callback([{ target: { getBoundingClientRect: jest.fn() } }]);
    }),
    disconnect: jest.fn(),
  }
)));

jest.mock('debounce', () => fn => () => fn());

it('should render if children is a component', () => {
  const tree = renderer.create((
    <VisibleBoundingBox>
      <div>foo</div>
    </VisibleBoundingBox>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should render if children is a function', () => {
  const bar = () => <p>bar</p>;
  const tree = renderer.create((
    <VisibleBoundingBox>
      {bar}
    </VisibleBoundingBox>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call onChange', () => {
  const onChange = jest.fn();
  shallow(
    <VisibleBoundingBox onChange={onChange}>
      <div>foo</div>
    </VisibleBoundingBox>,
  );
  expect(onChange.toHaveBeenCalled);
});

it('should call disconnect when component will unmount', () => {
  const wrapper = shallow(
    <VisibleBoundingBox>
      <div>foo</div>
    </VisibleBoundingBox>,
  );
  wrapper.unmount();
});
