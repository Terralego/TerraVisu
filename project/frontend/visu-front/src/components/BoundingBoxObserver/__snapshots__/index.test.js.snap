// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render if children is a component 1`] = `
<div>
  <div>
    foo
  </div>
</div>
`;

exports[`should render if children is a function 1`] = `
<p>
  bar
</p>
`;
